OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -269,33 +269,35 @@ public string VoicePhraseForOrder(Actor self, Order order)
 		public bool HasSpace(int weight) { return totalWeight + reservedWeight + weight <= Info.MaxWeight; }
 		public bool IsEmpty(Actor self) { return cargo.Count == 0; }
 
-		public Actor Peek(Actor self) { return cargo.Peek(); }
+		public Actor Peek(Actor self) { return cargo.Last(); }
 
-		public Actor Unload(Actor self)
+		public Actor Unload(Actor self, Actor passenger = null)
 		{
-			var a = cargo.Pop();
+			passenger = passenger ?? cargo.Last();
+			if (!cargo.Remove(passenger))
+				return passenger;
```

Comment:
We return `passenger` both when the unload was successful and when it wasn't.  Can we return null instead for the latter case? Ie. make `Unload` return the passenger that was actually unloaded, and when none was return null.

Reply:
I think i'd prefer to throw an explicit exception over returning null and then almost certainly NREing somewhere less obvious.

Reply:
Replaced returning null passenger with NRE